### ISSUE TITLE
TrustedDevice fails because of JSONField

### ISSUE THREAD

**Issue by pmosse** [mkalioby/django-mfa2#70]
<img width="1017" alt="Captura de pantalla 2022-10-31 a las 19 59 07" src="https://user-images.githubusercontent.com/6408341/199088258-a54c7568-781f-482e-aa91-698e2a04be4d.png">

Hello!

I am using jsonfield 3.1.0,  Django 3.2.6 and Python 3.8. When trying to add a TrustedDevice (following the example code, already integrated into my application, I am getting this error.

**Comment by mkalioby** [comment]
Sorry, to hear that, I acknowledge the issue and will solve as soon as possible

**Comment by mkalioby** [comment]
Hello, I fixed the issue and I did some refurbishing for trusted device page like adding QR for link
You can find the fix in [version 2.7.0rc1](https://pypi.org/project/django-mfa2/2.7.0rc1/). 
Please validate it and let me know so it gets released.

**Comment by pmosse** [comment]
Thank you for the fast reply! Now when I try to add a TrustedDevice it says I can't add more devices and I should delete the previous one (but I don't actually have any). When I refresh the page, I see it created 2 Trusted Devices for me...

A separated question: does this support Face ID in iOS?

Thank you!

**Comment by mkalioby** [comment]
> Thank you for the fast reply! Now when I try to add a TrustedDevice it says I can't add more devices and I should delete the previous one (but I don't actually have any). When I refresh the page, I see it created 2 Trusted Devices for me...

These are the failed registrations so you can  delete them and try again.
> 
> A separated question: does this support Face ID in iOS?

Not trusted device, but FIDO2 which is supported by the library does and it even supports pasakeys
> 
> Thank you!

**Comment by pmosse** [comment]
Great! But the failed registrations are created despite I get a message saying that I can't add more devices...
So it goes like this: I don't have any devices registered, when I try to register one it says I can't register more... and it creates these failed devices...

**Comment by mkalioby** [comment]
Did you add any other 2FA method before trusted device?

**Comment by pmosse** [comment]
Yes, just one FIDO2

**Comment by mkalioby** [comment]
Did you login with fido2 before registering trusted device enrollment?

**Comment by pmosse** [comment]
Yes, I did. I mean once I activate 2FA for a user (in this case, FIDO2), I will always login with it, right? I am doing the tests with a single user which has FIDO2 activated. But if your question is if I used it immediately before, I didn't. I just opened the browser in my mobile and it remembered me.

**Comment by mkalioby** [comment]
So is it working?

**Comment by pmosse** [comment]
No no, same situation as I explained before. Adding a trusted device fails but at the same time it creates 2 invalid records. If you send me a Telegram user or email I can share a video

**Comment by mkalioby** [comment]
Please make sure of the following before adding a trusted device
- you have another 2FA registered
- login with that factor and then try to register a trusted device

Github allows videos in comments now, btw.

**Comment by pmosse** [comment]
I followed those steps but I get the exact same result, unfortunately. About the video, I prefer not to share it publicly as it has company info. If you want to check it you can write me here https://privnote.com/S27n7jpS#CJXqXiFIP, otherwise we can continue here :)
